core/src/main/java/org/infinispan/metadata/EmbeddedMetadata.java

```diff
@@ -160,7 +160,7 @@ private EmbeddedExpirableMetadata(
             long lifespan, TimeUnit lifespanUnit,
             long maxIdle, TimeUnit maxIdleUnit, EntryVersion version) {
          super(version);
-         this.lifespan = lifespanUnit.toMillis(lifespan);
+         this.lifespan = lifespan < 0 ? lifespan : lifespanUnit.toMillis(lifespan);
          this.maxIdle = maxIdleUnit.toMillis(maxIdle);
```

Comment:
@vjuranek , please note there is another potential NPE in the toMillis() conversion in AbstractEmbeddedTimeoutMetadata.